gen/src/generate/ucd/mod.rs

```diff
@@ -0,0 +1,47 @@
+mod shared;
+
+mod age;
+mod bidi;
+mod category;
+mod core;
+mod normal;
+
+use std::io;
+
+pub use self::shared::unicode_data::{UnicodeData, UnicodeDataEntry};
+pub use self::shared::version::UnicodeVersion;
+
+pub fn generate_all<I, S>(crates: I) -> io::Result<()>
+where
+    I: Iterator<Item = S>,
+    S: AsRef<str>,
+{
+    println!(">>> Loading UCD Version");
+    let ucd_version = shared::version::read_unicode_version()?;
+    println!(">>> Loading UCD UnicodeData");
+    let unicode_data = shared::unicode_data::read_unicode_data()?;
+
+    for krate in crates {
+        let path = super::tables_path(krate.as_ref());
+        match krate.as_ref() {
+            "unic-ucd-age" => {
+                age::generate(path, &ucd_version, &unicode_data)?;
```

Comment:
Yeah, I missed the filtering based on CLI options. If we're keeping that, it's fine then.

In the Python scripts, I intentionally didn't make it optional to generate or download only one of the tables (or anything less than all the tables) of one source, to, in some level, enforce consistency between generated files. So, for each source, all data is download at the same time (old versions `rm`ed first), and all tables generated also at the same time.

How do you feel about that approach? Do you think the time it takes to re-generate the tables could become a problem in larger components and worth loosing the enforcement?